src/ml/neural_net/tf_model_backend.cpp

```diff
@@ -74,7 +72,30 @@ PYBIND11_MODULE(libtctensorflow, m) {
       });
 }
 
-tf_model_backend::tf_model_backend(pybind11::object model): model_(model) {}
+
+tf_model_backend::tf_model_backend(int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
+    const float_array_map& config, const float_array_map& weights, std::string model_name) {
+
+  if (model_name == "activity_classifier") {
```

Comment:
I don't think it makes sense to handle AC specific code here. I would either go back to the way you had it before, or go all the way forward to passing just the model name and all the other parameters (including batch size, input shape, config, weights) in a dictionary to a generic (non-AC-specific) Python module that handles everything AC-specific.

Offhand, for now I would go back to the way we had it before. I think it'll be more work than we want to do now to make this clean.

Reply:
Changed this